### ISSUE TITLE
ansible-doc should ignore modules with missing docs and with non-python extensions

### ISSUE THREAD

**Issue by gdahlm** [ansible/ansible#12516]
running `ansible-doc -l` when there are markdown files and/or modules with bad documentation breaks the command and results in a bad user experience.

As an example running `git clone git://github.com/ansible/ansible.git --recursive` on  commit `82b33c381f110e3bfa82dd589ad6c06fbd6be374` will require you to delete multiple readme files.

``` bash
gdah-mac:upstream gdah$ . ansible/hacking/env-setup
running egg_info
creating lib/ansible.egg-info
writing requirements to lib/ansible.egg-info/requires.txt
writing lib/ansible.egg-info/PKG-INFO
writing top-level names to lib/ansible.egg-info/top_level.txt
writing dependency_links to lib/ansible.egg-info/dependency_links.txt
writing manifest file 'lib/ansible.egg-info/SOURCES.txt'
reading manifest file 'lib/ansible.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
no previously-included directories found matching 'v2'
no previously-included directories found matching 'docsite'
no previously-included directories found matching 'ticket_stubs'
no previously-included directories found matching 'packaging'
no previously-included directories found matching 'test'
no previously-included directories found matching 'hacking'
no previously-included directories found matching 'lib/ansible/modules/core/.git'
no previously-included directories found matching 'lib/ansible/modules/extras/.git'
writing manifest file 'lib/ansible.egg-info/SOURCES.txt'

Setting up Ansible to run out of checkout...

PATH=/Users/gdah/build/upstream/ansible/bin:/opt/local/bin:/opt/local/sbin:/Users/gdah/.rvm/gems/ruby-2.0.0-p643/bin:/Users/gdah/.rvm/gems/ruby-2.0.0-p643@global/bin:/Users/gdah/.rvm/rubies/ruby-2.0.0-p643/bin:/opt/local/bin:/opt/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/MacGPG2/bin:/Users/gdah/.rvm/bin:/opt/local/Library/Frameworks/Python.framework/Versions/2.7/bin
PYTHONPATH=/Users/gdah/build/upstream/ansible/lib:
MANPATH=/Users/gdah/build/upstream/ansible/docs/man:

Remember, you may wish to specify your host file with -i

Done!

gdah-mac:upstream gdah$ ansible-doc -l
 [ERROR]: unable to parse /Users/gdah/build/upstream/ansible/lib/ansible/modules/core/CONTRIBUTING.md

ERROR! module CONTRIBUTING has a documentation error formatting or is missing documentation
```

Deleting the following files results in expected behavior.

```
ansible/lib/ansible/modules/core/CONTRIBUTING.md
ansible/lib/ansible/modules/extras/CONTRIBUTING.md
ansible/lib/ansible/modules/extras/cloud/amazon/GUIDELINES.md
ansible/lib/ansible/modules/core/README.md
ansible/lib/ansible/modules/core/cloud/openstack/README.md
ansible/lib/ansible/modules/extras/README.md
ansible/lib/ansible/modules/extras/REVIEWERS.md
ansible/lib/ansible/modules/core/cloud/rackspace
ansible/lib/ansible/modules/core/test-docs.sh
ansible/lib/ansible/modules/extras/test-docs.sh
```

It would be a better user experience to only display errors, but to continue on if the command is marked as verbose and to ignore non-doc plugins.

**Comment by gdahlm** [comment]
Adding full version information

```bash
$ ansible --version
ansible 2.0.0 (devel 82b33c381f) last updated 2015/09/24 12:31:10 (GMT -700)
  lib/ansible/modules/core: (detached HEAD e8227dea7a) last updated 2015/09/24 12:31:21 (GMT -700)
  lib/ansible/modules/extras: (detached HEAD f2faf70996) last updated 2015/09/24 12:31:30 (GMT -700)
  config file = /Users/gdah/.ansible.cfg
  configured module search path = None
```

**Comment by bcoca** [comment]
fixed via 6c190aa8a427ba692a6559fa9fe8f640107b590a, we now ignore all .md files and test-docs.sh
